portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -112,15 +124,72 @@ function insertSearch() {
 
   const searchInput = document.createElement('input');
   searchInput.className = 'form-control form-inline';
+  searchInput.id = 'userSearch';
   searchInput.type = 'text';
   searchInput.placeholder = 'Search';
 
   const searchI = document.createElement('i');
   searchI.className = 'fas fa-search text-white ml-3 mr-auto';
 
+  const searchResults = document.createElement('ul');
+  searchResults.id = 'searchResults';
+
   searchDiv.appendChild(searchInput);
   searchDiv.appendChild(searchI);
   searchBar.appendChild(searchDiv);
+  searchBar.append(searchResults);
+
+  return searchBar;
+}
+
+// searches each child Node of the page in the elementArray and retains the
+// elements that contain the wanted word
+function searchPages(elementArray) {
+  const wantedWords = document.getElementById('userSearch').value.toLowerCase();
+  const arrayLength = elementArray.length;
+
+  const resultArray = [];
+  const backArray = [];
+  if (wantedWords.length > 0) {
```

Comment:
@kennzoid comment:
https://github.com/kofimeighan/adelardos-capstone-2020/pull/17#discussion_r450623238
nit: you can make the code a little cleaner if you return immediately when this condition is not met.

if (wantedWords.length == 0) {
  return [];
}
Then the for loop, which is most of the function, doesn't need to be nested.

Reply:
Yeah that makes sense. Thanks Kenneth